src/main/kotlin/fi/metatavu/pakkasmarja/services/erp/sap/ContractsController.kt

```diff
@@ -351,38 +353,27 @@ class ContractsController: AbstractSapResourceController() {
      * @return item codes
      */
     private fun getGroupItemCodes(groupCode: Int, sapSession: SapSession): List<String> {
-        val filter = constructGroupCodeFilter(groupCode)
-        val select = "\$select=ItemCode"
-
-        val requestUrl = constructSAPRequestUrl(
-            baseUrl = "${sapSession.apiUrl}/Items",
-            select = select,
-            filter = filter,
-            firstResult = null,
-            maxResults = null
-        )
-
-        val itemsResponse = sapListItemsRequest(
-            requestUrl = requestUrl,
-            sapSession = sapSession
-        ) ?: return emptyList()
-
-        val items: List<Item> = itemsResponse
-        return items.map{ item -> item.itemCode }
+        return itemsController.listItems(
+            sapSession = sapSession,
+            itemGroupCode = groupCode,
+            updatedAfter = null
+        ).map(Item::itemCode)
     }
 
     /**
-     * Constructs group code filter for contract listing
+     * Reads doc nummer from sap contract
      *
-     * @param code group code
-     * @return constructed filter string
+     * @param sapContract SAP contract
+     * @return doc number
      */
-    private fun constructGroupCodeFilter(code: Int): String {
-        val groupCode = configController.getGroupPropertiesFromConfigFile().find { groupCode -> groupCode.code == code } ?: return ""
-        val isOrganicFilter = "Properties21 eq SAPB1.BoYesNoEnum'${toSapItemPropertyBoolean(groupCode.isOrganic)}'"
-        val isFrozenFilter = "Properties28 eq SAPB1.BoYesNoEnum'${toSapItemPropertyBoolean(groupCode.isFrozen)}'"
+    private fun getDocNum(sapContract: SapContract): Int? {
+        val id = sapContract.id ?: return null
+        val parts = id.split("-")
+        if (parts.size != 2) {
```

Comment:
Length should be 3, since sapContract ID should be formed like
`[year]-[DocNum]-[ItemGroupCode]`, e.g.
`2021-1070-105`